src/commands/cem_progress.ts

```diff
@@ -0,0 +1,151 @@
+import { app } from '../initializers/bolt'
+import { firestore } from '../initializers/firebase'
+import { Modal, Message } from '../types/slack'
+
+app.command(`/cem_progress`, async ({ payload, ack, context }) => {
+  ack()
+  const challengerRef = firestore.collection(`challengers`).doc(payload.user_id)
+  const projectsRef = firestore.collection(`projects`)
+  const projectsQuery = projectsRef
+    .where(`challenger`, `==`, challengerRef)
+    .where(`status`, `==`, `published`)
+
+  const projects = await projectsQuery.get().catch(err => {
+    throw new Error(err)
+  })
+
+  if (projects.docs.length === 0) {
+    const msg: Message = {
+      token: context.botToken,
+      text: `中間報告できるプロジェクトはありません`,
+      channel: payload.channel_id,
+      user: payload.user_id,
+    }
+    return app.client.chat.postEphemeral(msg as any)
+  }
+
+  // let index = 0
+  const blocks: any[] = []
+  // async/awaitを使いたいので、for-ofを使用している
+  for (const project of projects.docs) {
+    const challlengeRef = projectsRef.doc(project.id).collection(`challenges`)
+    const challenges = await challlengeRef.get()
+
+    const projData = project.data()
+    blocks.push({
+      type: `section`,
+      text: {
+        type: `plain_text`,
+        text: `プロジェクト名：${projData.title}`,
+        emoji: true,
+      },
+    })
+
+    challenges.docs.forEach(challenge => {
+      const challengeData = challenge.data()
+      blocks.push({
+        type: `section`,
+        text: {
+          type: `mrkdwn`,
+          text: `挑戦：${challengeData.name}`,
+        },
+        accessory: {
+          action_id: `progress_${project.ref.id}_${challenge.ref.id}`,
+          type: `static_select`,
+          placeholder: {
+            type: `plain_text`,
+            text: `進捗を選択`,
+          },
+          options: [
+            {
+              text: {
+                type: `plain_text`,
+                text: `未着手`,
+              },
+              value: `notStarted`,
+            },
+            {
+              text: {
+                type: `plain_text`,
+                text: `進捗半分以下`,
+              },
+              value: `lessHalf`,
+            },
+            {
+              text: {
+                type: `plain_text`,
+                text: `進捗半分以上`,
+              },
+              value: `overHalf`,
+            },
+            {
+              text: {
+                type: `plain_text`,
+                text: `完了済`,
+              },
+              value: `completed`,
+            },
+          ],
+        },
+      })
+
+      blocks.push({
+        type: `input`,
+        block_id: `comment_${project.ref.id}_${challenge.ref.id}`,
+        element: {
+          type: `plain_text_input`,
+          // multiline: true,
+          initial_value: projData.description,
+          action_id: `comment`,
+          placeholder: {
+            type: `plain_text`,
+            text: `挑戦に対するコメント`,
+          },
+        },
```

Comment:
コメントはオプショナルのほうが使い勝手良いかもしれませんね

```suggestion
        },
        optional: true,
```

Reply:
option化しました